### PR TITLE
[WIP] Enh vizu

### PR DESCRIPTION
**Pull request by dohmatob**
The idea is to be able to obtain detailed plots like this:

keras_mnist_nnet

### CODE REVIEW

**Comment by tboquet** [comment]
Nice addition!
For the documentation, Keras uses its own markup parser. I'm not sure if the numpy like syntax will be nicely parsed. You can modify the doc like this:

```
"""
# Parameters
    verbose: boolean, optional (default False). If true, then more layer parameters will be reported in the generated plot.
...
"""
```

**Comment by dohmatob** [comment]
@tboquet: Thanks for the suggestion. I've integrated it.

**Comment by hadim** [comment]
Any news for this that would be a really nice feature !

File: keras/utils/visualize_util.py [hunk 1]
```diff
@@ -9,7 +9,15 @@
                        ' and graphviz for `pydotprint` to work.')
 
 
-def model_to_dot(model, show_shapes=False, show_layer_names=True):
+def model_to_dot(model, show_shapes=False, show_layer_names=True,
+                 verbose=False):
```
Comment (fchollet) [inline keras/utils/visualize_util.py]: `verbose` generally means something else (verbosity level in logs). Rename to `show_layer_attributes`.

File: keras/utils/visualize_util.py [hunk 2]
```diff
@@ -43,7 +51,43 @@ def model_to_dot(model, show_shapes=False, show_layer_names=True):
                     [str(ishape) for ishape in layer.input_shapes])
             else:
                 inputlabels = 'multiple'
-            label = '%s\n|{input:|output:}|{{%s}|{%s}}' % (label, inputlabels, outputlabels)
+
+            def _get_val(stuff):
+                if hasattr(stuff, "__name__"):
```
Comment (fchollet) [inline keras/utils/visualize_util.py]: You make use of  `"`  but the file uses `'`. Normalize to `'`.

File: keras/utils/visualize_util.py [hunk 3]
```diff
@@ -43,7 +51,43 @@ def model_to_dot(model, show_shapes=False, show_layer_names=True):
                     [str(ishape) for ishape in layer.input_shapes])
             else:
                 inputlabels = 'multiple'
-            label = '%s\n|{input:|output:}|{{%s}|{%s}}' % (label, inputlabels, outputlabels)
+
+            def _get_val(stuff):
```
Comment (fchollet) [inline keras/utils/visualize_util.py]: `stuff` is not an appropriate variable name.

**Comment by fchollet** [comment]
Closing outdated PR. If you still care about the content of the PR, please submit a new PR to `master`, updated for the Keras 2.0 API.

**Pull request closed without merging**